### PR TITLE
SAK-44225 SakaiGrader keep the selected submission

### PR DESCRIPTION
**Pull request by st-manu**
After thoroughly testing the previous PR, I found that with the change, the selected student is lost, and the first one is set by default. Similarly (as was already happening), when changing groups, the displayed student is not retained if they are in the group being filtered.

### CODE REVIEW

File: webcomponents/tool/src/main/frontend/packages/sakai-grader/src/SakaiGrader.js [hunk 1]
```diff
@@ -672,14 +673,25 @@ export class SakaiGrader extends graderRenderingMixin(gradableDataMixin(SakaiEle
...
     if (filtered.length > 0) {
-      const firstSubmissionId = filtered[0].id;
-      this._hydrateCluster(firstSubmissionId).then(submission => {
-
-        if (submission) {
-          this._submissions = [ ...filtered ];
-          this._submission = submission;
-        }
-      });
+      const currentSubmissionId = this._submission.id;
```
[the remark this answers is not in the document]
Comment (st-manu) [reply webcomponents/tool/src/main/frontend/packages/sakai-grader/src/SakaiGrader.js:676]: @adrianfish thank you for your comment. I'm updating the commit.

File: webcomponents/tool/src/main/frontend/packages/sakai-grader/src/SakaiGrader.js [hunk 2]
```diff
@@ -672,14 +673,24 @@ export class SakaiGrader extends graderRenderingMixin(gradableDataMixin(SakaiEle
...
-      const firstSubmissionId = filtered[0].id;
-      this._hydrateCluster(firstSubmissionId).then(submission => {
+      const currentSubmission = filtered.find(s => s.id === this._submission.id);
 
-        if (submission) {
-          this._submissions = [ ...filtered ];
-          this._submission = submission;
-        }
-      });
+      if (currentSubmission) {
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/packages/sakai-grader/src/SakaiGrader.js:678]:
```suggestion
      if (filtered.some(s => s.id === this._submission.id)) {
```

Comment (adrianfish) [reply webcomponents/tool/src/main/frontend/packages/sakai-grader/src/SakaiGrader.js:678]: No need for the variable at all :)

**Review by adrianfish — CHANGES_REQUESTED** [review]
Just a small change :)

File: webcomponents/tool/src/main/frontend/packages/sakai-grader/src/SakaiGrader.js [hunk 3]
```diff
@@ -672,14 +673,24 @@ export class SakaiGrader extends graderRenderingMixin(gradableDataMixin(SakaiEle
     }
 
     if (filtered.length > 0) {
-      const firstSubmissionId = filtered[0].id;
-      this._hydrateCluster(firstSubmissionId).then(submission => {
+      const currentSubmission = filtered.find(s => s.id === this._submission.id);
```
Comment (adrianfish) [inline webcomponents/tool/src/main/frontend/packages/sakai-grader/src/SakaiGrader.js:676]:
```suggestion
```

**Comment by st-manu** [comment]
@adrianfish thank you for the suggestions!

**Review by adrianfish — APPROVED** [review]
Thanks for this work. It's good to see other people doing work in the grader. It's a complicated piece of code.

**Pull request merged**